### PR TITLE
feat(cli): Add asset selector for `wallet:send`

### PR DESCRIPTION
**Pull request by mat-if**
## Summary

Per the [oclif docs](https://oclif.io/docs/prompting#inquirer), they recommend the `inquirer` lib for this. We also had it installed already as a transitive dependency.

Since we're adding this new functionality: we should use this `inquirer` selector any time we have a known, finite list of things to select from. It's a huge improvement over making the user go look up the information out-of-band and then look up the flag to pass into the CLI.

Screen Recording 2023-01-13 at 10 30 31 AM

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -7040,30 +7054,10 @@ init-package-json@^3.0.2:
     validate-npm-package-license "^3.0.4"
     validate-npm-package-name "^4.0.0"
 
-inquirer@^8.0.0:
```
Comment (mat-if) [inline yarn.lock:7043]: I got a little fiddly here to de-dupe these transitive dependencies, so we have 1 less thing downloaded/installed.

**Review by jowparks — APPROVED** [review]

File: ironfish-cli/src/commands/wallet/send.ts [hunk 2]
```diff
@@ -265,4 +278,44 @@ Find the transaction on https://explorer.ironfish.network/transaction/${
...
+          name: `${assetId} (${displayName})`,
+        })
+      }
+    }
+
+    let assetId
+
+    if (balancesResponse.content.balances.length === 1) {
+      // If there's only one available asset, showing the choices is unnecessary
+      assetId = balancesResponse.content.balances[0].assetId
```
Comment (hughy) [inline ironfish-cli/src/commands/wallet/send.ts:306]: We could check this condition before the loop above to avoid an unnecessary `getAsset` call

Comment (hughy) [reply ironfish-cli/src/commands/wallet/send.ts:306]: I'm also thinking that in practice, if there's only one asset, doesn't it have to be $IRON in order to send a transaction? I don't think that we should hardcode that, but `send` might always fail if there's no iron for a tx fee

**Review by hughy — APPROVED** [review]
The selector looks great!

File: ironfish-cli/src/commands/wallet/send.ts (hunk 2 above)
Comment (mat-if) [reply ironfish-cli/src/commands/wallet/send.ts]: That's a good point. I think we could improve the command's usability further with more guardrails. For example, the send command will happily let you start the command even if you have 0 balance, which can be confusing and strange. 

Also per the first comment, moved this up.

**Review by NullSoldier — APPROVED** [review]

File: ironfish-cli/src/commands/wallet/send.ts [hunk 3]
```diff
@@ -265,4 +278,43 @@ Find the transaction on https://explorer.ironfish.network/transaction/${
...
+        })
+      }
+    }
+
+    const response: { assetId: string } = await inquirer.prompt<{ assetId: string }>([
+      {
+        name: 'assetId',
+        message: 'Select the asset you wish to send',
+        type: 'list',
+        choices: assetOptions,
```
Comment (NullSoldier) [inline ironfish-cli/src/commands/wallet/send.ts:315]: It would be nice to sort the native asset to the first option

**Pull request merged**